### PR TITLE
all: Clean up a bunch of unused code

### PR DESCRIPTION
**Pull request by Jacalz**
This cleans up some unused code as a step in the right direction for #18. I assume that some of the things that I have removed might be accidentally unused or just intended to be kept around for some debug purposes. Please have a careful look that nothing is deleted that you wish to keep. Tests all still pass.

I highly recommend that you run `staticcheck ./...` on the repository to address reported issues and clean up unused code and also large portions of commented out code. This would make the codebase a lot clearer.

### CODE REVIEW

**Comment by Jacalz** [comment]
For Fyne binaries this seems to decrease the size about 0.1MB while memory usage is about the same. There is more work to be done here but this should be a good start.

**Comment by benoitkugler** [comment]
Thank you, I'll look at it as soon as possible.

**Comment by benoitkugler** [comment]
I've reviewed staticcheck warnings and removed almost all dead code, keeping a few functions which may be useful in the future.
The resulting PR is at #22. I'll close this one, feel free to re-open it if you think I've missed something.

**Comment by Jacalz** [comment]
Thanks. I will have a look later today. Any chance that you could revert back to using `ioutil`? We are using Go 1.14 as minimum version in Fyne and this change would prohibit us from upgrading 😅

**Comment by Jacalz** [comment]
The revert of ioutil removal seems to have reverted all of the changes. Was that intentional?
https://github.com/benoitkugler/textlayout/commit/38aec5f3bd2c41acf1d01bb94291d81aad76dd39

**Comment by Jacalz** [comment]
Never mind. I think the git history is a bit hard to follow. The code looks fine on my end

**Comment by benoitkugler** [comment]
> Never mind. I think the git history is a bit hard to follow. The code looks fine on my end

My mistake here... (I got confused when reverting the last commit..)
The last version (tag v0.2.3) should be fine indeed.
